nexus/src/app/mod.rs

```diff
@@ -245,6 +297,56 @@ impl Nexus {
         }
     }
 
+    // Called to hand off management of external servers to Nexus.
+    pub(crate) async fn set_servers(
+        &self,
+        external_servers: ExternalServers,
+        internal_server: DropshotServer,
+    ) {
+        // If any servers already exist, close them.
+        let _ = self.close_servers().await;
+
+        // Insert the new servers.
+        *self.external_servers.lock().await = external_servers;
+        self.internal_server.lock().await.replace(internal_server);
+    }
+
+    pub async fn close_servers(&self) -> Result<(), String> {
+        let mut external_servers = self.external_servers.lock().await;
+        if let Some(server) = external_servers.http.take() {
+            server.close().await?;
+        }
+        if let Some(server) = external_servers.https.take() {
+            server.close().await?;
+        }
+        if let Some(server) = self.internal_server.lock().await.take() {
+            server.close().await?;
+        }
+        Ok(())
+    }
+
+    pub async fn get_http_external_server(
+        &self,
+    ) -> Option<std::net::SocketAddr> {
+        let external_servers = self.external_servers.lock().await;
+        external_servers.http.as_ref().map(|server| server.local_addr())
+    }
+
+    pub async fn get_https_external_server(
+        &self,
+    ) -> Option<std::net::SocketAddr> {
+        let external_servers = self.external_servers.lock().await;
+        external_servers.https.as_ref().map(|server| server.local_addr())
+    }
+
+    pub async fn get_internal_server(&self) -> Option<std::net::SocketAddr> {
+        self.internal_server
+            .lock()
+            .await
+            .as_ref()
+            .map(|server| server.local_addr())
+    }
```

Comment:
These seem to return the external server's address rather than a handle to the server itself, as the names suggest? Maybe `get_http_external_server_address` etc?

Reply:
Done in 8450dbb1aaed31ac1182055eebc42406704b3b99